crates/swc_ecma_transforms_proposal/src/decorator_2022_03.rs

```diff
@@ -421,289 +421,273 @@ impl Decorator202203 {
         new_class_name
     }
 
-    fn handle_class_decl(&mut self, c: &mut ClassDecl) -> Option<Stmt> {
+    // This function will call `visit` internally.
+    fn handle_class_decl(&mut self, c: &mut ClassDecl) -> Stmt {
```

Comment:
The git diff is misleading. 
There are minimal logical changes within this function.

1. Change `Option<Stmt>` to `Stmt`
2. All `class.decorators.is_empty` checks have been moved to the caller side.